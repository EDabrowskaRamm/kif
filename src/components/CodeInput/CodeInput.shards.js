import styled from '@emotion/styled'

export const InputWrapper = styled.div`
  position: relative;
  font-style: normal;
  font-weight: bold;
  font-size: 1.6rem;
  line-height: 2.4rem;
  margin: 40px auto 20px;
  max-width: 290px;
`

export const Error = styled.span`
  position: absolute;
  bottom: 39%;
  left: 50%;
  transform: translateX(-50%);
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.6rem;
`