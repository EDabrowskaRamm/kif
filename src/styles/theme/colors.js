const _white = '#FFF'
const _colorGrey47 = '#707181'

const _tuna = '#30313D'
const _stormGray = _colorGrey47
const _snuff = '#DBDCEE'
const _luckyPoint = '#211965'
const _bittersweet = '#FF7165'

export const background = {
  primary: _white,
  overlay: _tuna,
}

export const text = {
  primary: _stormGray,
  secondary: _white,
  title: _tuna,
  error: _bittersweet,
}

export const border = {
  input: _snuff,
  error: _bittersweet,
}

export const buttonActive = {
  text: _white,
  background: _bittersweet
}

export const buttonDisabled = {
  text: _luckyPoint,
  background: 'rgba(141, 142, 161, 0.1)'
}