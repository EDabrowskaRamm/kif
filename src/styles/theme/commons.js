import { border } from './colors'

export const borders = {
  input: `1px solid ${border.input}`,
  error: `1px solid ${border.error}`,
}

export const borderRadius = {
  button: '4px',
  input: '10px',
  modal: '16px'
}