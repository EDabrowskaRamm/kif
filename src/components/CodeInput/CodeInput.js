import React from 'react'
import PinInput from 'react-pin-input'

import { InputWrapper, Error } from './CodeInput.shards'
import { borders, borderRadius } from '../../styles/theme/commons'
import { Button } from '../Button/Button'

const CodeInput = ({
  handleChange,
  handleConfirm,
  error = false,
  active = false
}) => {
  return <InputWrapper>
    <PinInput
      length={4}
      initialValue=''
      type='numeric'
      onChange={handleChange}
      style={{
        display: 'flex',
        justifyContent: 'space-evenly',
        marginBottom: '40px'
      }}
      inputStyle={{
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        border: error ? borders.error : borders.input,
        borderRadius: borderRadius.input,
        padding: '22px 18px',
        width: '50px',
        height: '68px',
      }}
    />
    {error && <Error>Kod PIN musi zawierać 4 cyfry</Error>}
    <Button
      handleClick={handleConfirm}
      text='Potwierdź'
      active={active}
    />
  </InputWrapper>
}

export default CodeInput