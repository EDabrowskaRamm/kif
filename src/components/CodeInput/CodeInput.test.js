import React from 'react'
import renderer from 'react-test-renderer'
import serializer from 'jest-emotion'
import { ThemeProvider } from 'emotion-theming'

import CodeInput from './CodeInput'

import theme from '../../styles/theme/theme'

expect.addSnapshotSerializer(serializer)

describe('renders CodeInput', () => {
  it('Snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <CodeInput handleChange={() => { }} handleConfirm={() => { }} />
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
