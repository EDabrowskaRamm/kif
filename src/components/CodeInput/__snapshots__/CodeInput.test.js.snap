// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CodeInput Snapshot 1`] = `
.emotion-1 {
  position: relative;
  font-style: normal;
  font-weight: bold;
  font-size: 1.6rem;
  line-height: 2.4rem;
  margin: 40px auto 20px;
  max-width: 290px;
}

.emotion-0 {
  border-radius: 4px;
  background-color: #FF7165;
  color: #FFF;
  padding: 16px 36px;
  outline: none;
  cursor: pointer;
  font-style: normal;
  font-weight: bold;
  font-size: 1.4rem;
  line-height: 1.6rem;
}

.emotion-0:disabled {
  color: #211965;
  background: rgba(141,142,161,0.1);
  cursor: default;
}

@media (min-width:992px) {
  .emotion-0 {
    font-size: 1.6rem;
  }
}

<div
  className="emotion-1"
>
  <div
    className="pincode-input-container"
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-evenly",
        "marginBottom": "40px",
      }
    }
  >
    <input
      aria-label=""
      autoComplete="off"
      className="pincode-input-text"
      maxLength="1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      pattern="[0-9]*"
      placeholder=""
      style={
        Object {
          "alignItems": "center",
          "background": "transparent",
          "border": "1px solid #DBDCEE",
          "borderRadius": "10px",
          "display": "flex",
          "height": "68px",
          "justifyContent": "center",
          "margin": "0 2px",
          "padding": "22px 18px",
          "textAlign": "center",
          "width": "50px",
        }
      }
      type="tel"
      value=""
    />
    <input
      aria-label=""
      autoComplete="off"
      className="pincode-input-text"
      maxLength="1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      pattern="[0-9]*"
      placeholder=""
      style={
        Object {
          "alignItems": "center",
          "background": "transparent",
          "border": "1px solid #DBDCEE",
          "borderRadius": "10px",
          "display": "flex",
          "height": "68px",
          "justifyContent": "center",
          "margin": "0 2px",
          "padding": "22px 18px",
          "textAlign": "center",
          "width": "50px",
        }
      }
      type="tel"
      value=""
    />
    <input
      aria-label=""
      autoComplete="off"
      className="pincode-input-text"
      maxLength="1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      pattern="[0-9]*"
      placeholder=""
      style={
        Object {
          "alignItems": "center",
          "background": "transparent",
          "border": "1px solid #DBDCEE",
          "borderRadius": "10px",
          "display": "flex",
          "height": "68px",
          "justifyContent": "center",
          "margin": "0 2px",
          "padding": "22px 18px",
          "textAlign": "center",
          "width": "50px",
        }
      }
      type="tel"
      value=""
    />
    <input
      aria-label=""
      autoComplete="off"
      className="pincode-input-text"
      maxLength="1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      pattern="[0-9]*"
      placeholder=""
      style={
        Object {
          "alignItems": "center",
          "background": "transparent",
          "border": "1px solid #DBDCEE",
          "borderRadius": "10px",
          "display": "flex",
          "height": "68px",
          "justifyContent": "center",
          "margin": "0 2px",
          "padding": "22px 18px",
          "textAlign": "center",
          "width": "50px",
        }
      }
      type="tel"
      value=""
    />
  </div>
  <button
    className="emotion-0"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Potwierd≈∫
  </button>
</div>
`;
